mono/metadata/marshal-ilgen.c

```diff
@@ -401,19 +401,19 @@ emit_ptr_to_object_conv (MonoMethodBuilder *mb, MonoType *type, MonoMarshalConv
 		mono_mb_emit_byte (mb, CEE_LDIND_REF);
 		mono_mb_emit_ldloc (mb, 0);
 		mono_mb_emit_icon (mb, mspec->data.array_data.num_elem);
-		mono_mb_emit_icall (mb, mono_byvalarray_to_byte_array);
+		mono_mb_emit_icall (mb, mono_byvalarray_to_byte_array_raw);
```

Comment:
I wonder about the usability of this approach.  So I say `MONO_HANDLE_REGISTER_ICALL (mono_byvalarray_to_byte_array, ...)` and then I get to define `mono_byvalarray_to_byte_array`, but then when I go to write a _call_ I have to write `mono_byvalarray_to_byte_array_raw` or else I will get mysterious failures.

Maybe it's better if `mono_byvalarray_to_byte_array` was the name of the wrapper, and the implementation went into `mono_byvalarray_to_byte_array_impl` or something like that.

At least in that case if I make a mistake and write my own definition of `mono_byvalarray_to_byte_array`, I will get either a compile or link-time error, which seems better than a crash at runtime.

Does that make sense?